src/components/panels/shared.tsx

```diff
@@ -0,0 +1,15 @@
+type data = {
```

Comment:
You've been switching between the use of `type` and `interface` - while they are very similar we should probably just stick with one....my pick would be `interface` unless there is a reason to use `type`.

Reply:
After doing some more investigations, `type` might be the one to stick with unless we need the features from `interface` :)